### PR TITLE
Add support for Swift Package Manager

### PR DESCRIPTION
**Pull request by ikesyo**
Only `swift build` on macOS is supported atm. `swift test` and Linux platform are not supported yet.

### CODE REVIEW

File: .swift-version [hunk 1]
```diff
@@ -0,0 +1 @@
+3.0-PREVIEW-4
```
Comment (NachoSoto) [inline .swift-version]: Why not 5?

File: .travis.yml [hunk 2]
```diff
@@ -38,6 +38,11 @@ matrix:
         - carthage build --no-skip-current
       env:
         - JOB=CARTHAGE
+    - os: osx
+      language: generic
+      script: swift build
```
Comment (NachoSoto) [inline .travis.yml]: Awesome

**Comment by NachoSoto** [comment]
Perfect!

File: .swift-version (hunk 1 above)
Comment (ikesyo) [inline .swift-version]: Preview 5 is not released yet (this is mainly for Linux support preparation): https://swift.org/download/#previews 😅

**Pull request merged**